lib/vpc/manager/Manager.rb

```diff
@@ -0,0 +1,674 @@
+require "common/manager/Manager"
+require "conf/Configuration"
+require "util/Colors"
+require "vpc/loader/Loader"
+require "vpc/models/VpcDiff"
+require "vpc/models/RouteTableDiff"
+require "vpc/models/EndpointDiff"
+require "vpc/models/NetworkAclDiff"
+require "vpc/models/SubnetDiff"
+require "ec2/EC2"
+require "ec2/IPProtocolMapping"
+
+require "aws-sdk"
+require "json"
+
+module Cumulus
+  module VPC
+    class Manager < Common::Manager
+
+      include VpcChange
+
+      def initialize
+        super()
+        @create_asset = false
+        @client = Aws::EC2::Client.new(region: Configuration.instance.region, profile: Configuration.instance.profile)
+      end
+
+      def resource_name
+        "Virtual Private Cloud"
+      end
+
+      def local_resources
+        @local_resources ||= Hash[Loader.vpcs.map { |local| [local.name, local] }]
+      end
+
+      def aws_resources
+        @aws_resources ||= EC2::named_vpcs
+      end
+
+      def unmanaged_diff(aws)
+        VpcDiff.unmanaged(aws)
+      end
+
+      def added_diff(local)
+        VpcDiff.added(local)
+      end
+
+      def diff_resource(local, aws)
+        local.diff(aws)
+      end
+
+      def update(local, diffs)
+        aws_vpc = EC2::named_vpcs[local.name]
+
+        diffs.each do |diff|
+          case diff.type
+          when CIDR
+            puts "CIDR Block cannot be updated. You must create a new VPC."
+          when TENANCY
+            puts "Tenancy cannot be updated. You must create a new VPC."
+          when DHCP
+            puts "Updating DHCP Options..."
```

Comment:
Yeah I'll do that on the upper level ones but leave the lower level ones white since there can be a lot per section